### PR TITLE
[1 of 2] Add DOMUtil#closest

### PR DESCRIPTION
**Pull request by jfurrow**
Adds #closest from DOMUtil in [reactjs-components](https://github.com/mesosphere/reactjs-components/blob/master/src/Util/DOMUtil.js#L28-L40).

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @orlandohohmeier

File: src/js/utils/__tests__/DOMUtils-test.js [hunk 1]
```diff
@@ -4,6 +4,60 @@ var DOMUtils = require('../DOMUtils');
 
 describe('DOMUtils', function () {
 
+  describe('#closest', function () {
+
+    it('should should return the parent element when provided a selector and ' +
```
Comment (MatApple) [inline src/js/utils/__tests__/DOMUtils-test.js]: #nit but all these tests start with `should should`

File: src/js/utils/DOMUtils.js [hunk 2]
```diff
@@ -1,6 +1,26 @@
 const HEIGHT_ATTRIBUTES = ['paddingTop', 'paddingBottom', 'borderTopWidth', 'borderBottomWidth'];
 const WIDTH_ATTRIBUTES = ['paddingLeft', 'paddingRight', 'borderLeftWidth', 'borderRightWidth'];
 
+let matchesFn = function () {
```
Comment (MatApple) [inline src/js/utils/DOMUtils.js]: Another minor #nit but lets wrap this function in parenthesis so its easier to identify as a self executing function.

**Comment by kennyt** [comment]
LGTM after mat's comments

File: src/js/utils/__tests__/DOMUtils-test.js (hunk 1 above)
Comment (jfurrow) [inline src/js/utils/__tests__/DOMUtils-test.js]: Done!

File: src/js/utils/DOMUtils.js (hunk 2 above)
Comment (jfurrow) [inline src/js/utils/DOMUtils.js]: Done!

**Pull request merged**